src/metabase/automagic_dashboards/filters.clj

```diff
@@ -137,10 +137,11 @@
                                        (into {})))
         (add-filters dashboard max-filters)))
   ([dashboard dimensions max-filters]
-   (let [fks (->> (db/select Field
-                    :fk_target_field_id [:not= nil]
-                    :table_id [:in (keep (comp :table_id :card) (:ordered_cards dashboard))])
-                  field/with-targets)]
+   (let [fks (when-let [table-ids (not-empty (set (keep (comp :table_id :card)
+                                                        (:ordered_cards dashboard))))]
+               (->> (db/select Field :fk_target_field_id [:not= nil]
+                               :table_id [:in table-ids])
+                    field/with-targets))]
```

Comment:
This was broken if `table-ids` was an empty sequence -- `table_id IN ()` is not valid SQL. Fixed by not running the query at all in this case